text/002-css-rules.md

```diff
@@ -0,0 +1,987 @@
+- Start Date: 2024-02-24
+
+# Summary
+
+Styling for cases where props are required.  
+It will be partial coverage of [`@mixin` in SCSS](https://sass-lang.com/documentation/at-rules/mixin/) and [Variants in Stitches](https://stitches.dev/docs/variants).
+
+# Motivation
+
+- Complementing composition
+- Dynamic styling
+- Take advantage of preprocessors
+- Variants that correspond to semantic hierarchies
+- Improve [Vaniila extract's recipes](https://vanilla-extract.style/documentation/packages/recipes/)
+- Make your BEM expressive
+
+## Composition
+
+[css literals RFC](./000-css-literals.md#2-object-styles) is used to create each class.
+
+It's more of a feature for a single class.
+
+```tsx
+const mySample = css({ color: "red", padding: 3 });
+
+function Sample() {
+  return <div className={mySample}>contents...</div>;
+}
+```
+
+In reality, you'll likely use multiple classes.
+
+```tsx
+function Sample() {
+  const activate = useState(true);
+  return (
+    <div className={`class1 class2 ${activate && "class3"}`}>contents...</div>
+  );
+}
+```
+
+This is where using [clsx](https://github.com/lukeed/clsx) looks better.
+
+```tsx
+function Sample() {
+  const activate = useState(true);
+  return (
+    <div className={clsx(["class1", "class2", { class3: activate }])}>
+      contents...
+    </div>
+  );
+}
+```
+
+For the static case, you can leverage formal [compositing to reuse styles](./000-css-literals.md#14-css-composition) for optimization.
+
+```typescript
+const base = css({ padding: 12 });
+const primary = css([base, { background: "blue" }]);
+const secondary = css([base, { background: "aqua" }]);
+```
+
+**Compiled:**
+
+```css
+.[FILE_NAME]_base__[HASH] {
+  padding: 12px;
+}
+
+.[FILE_NAME]_primary__[HASH] {
+  background: blue;
+}
+
+.[FILE_NAME]_secondary__[HASH] {
+  background: aqua;
+}
+```
+
+## Dynamic Styling
+
+It's clear that composition alone can't handle all dynamic cases.  
+What about dynamic cases?
+
+The solution is known as [`UI = f(state)`](https://docs.flutter.dev/data-and-backend/state-mgmt/declarative) and is It's simple.
+
+If so, is there any reason for Style to not be a function?  
+This is the same as [fela's approach](https://fela.js.org/docs/latest/intro/principles).
+
+```typescript
+const rule = (props) => ({
+  fontSize: props.size + "px",
+  color: "blue",
+  lineHeight: props.lineHeight,
+  display: "flex",
```

Comment:
> [!NOTE]
> **P0[Critical defects]: There is a serious issue that requires further review and cannot be merged in its current state.**

I think it's IDE settings, but the code was automatically formatted.
In our code style, we don't have `,' at the end of an object.

Reply:
OK, i got it